### DISCUSSION TITLE
Show warning when importing swagger 2.0 files

### DISCUSSION THREAD

**Discussion by 0pilatos0** [scalar/scalar#3105] [Ideas]
I encountered an issue where I accidentally imported a Swagger 2.0 file into the Scalar OpenAPI client. This caused certain elements in my project to behave unexpectedly, which took me a significant amount of time to troubleshoot. After some investigation, I realized that the root of the problem was due to importing a Swagger 2.0 file instead of an OpenAPI 3.x file, which the client fully supports.

To improve the developer experience and prevent similar issues in the future, I would like to request the following enhancement:

Feature Request:

	•	Add a warning or notification when a Swagger 2.0 file is imported into the Scalar OpenAPI client. The warning should clearly inform the user that only OpenAPI 3.x files are supported and advise them to upgrade their specification to the correct version.

This will save time and avoid confusion for developers who might accidentally use the older version of Swagger when exporting or generating their API documentation.

**Comment by amritk** [comment]
Hey @0pilatos0 excellent idea! We have that done on the new refactor branch. Should get merged early next week!
